src/ansys/mapdl/core/cndparser.py

```diff
@@ -0,0 +1,248 @@
+    #Writing frequency of contact variables 
+    #FRQ = ["ITERATION","SUBSTEP","LOAD_STEP"]
+    #Name of the contact information for each pair 
+    ContNameHeader = []
+    #Name of the information for the each writing frequency
+    LoadTimeHeader = [ 'LoadStep', 'SubStep', 'IterationNo', 'Time', 'Scaled Time for Wear' ]
+    # Numeric variables
+    #Information per contact pair
+    nInfoPerPair = [] 
+    #Number of contact pair
+    paircount = 0  
+    #ContPairInfo[frq,ncontpair,:] = [ContPair ID, p2, p3, ... nInfoPerContPair] 
+    ContPairInfo = [] 
+    #LoadTime[frq,:] = [ LoadStep, SubStep, IterationNo, Time, Scaled Time for Wear]
+    LoadTime = []  
+    #reading the full file line by line   
+    cntr = 0
+    frq_num = 0
+    scaletimeflg = 0     
+    with open(filename, 'r') as reader:
+        for line in reader:
+            cntr = cntr + 1         
+            if cntr == 1:                     # <SOLUTION> 
+               pass 
+            elif cntr == 2:                   # <HEADER FRQ=...>
+               frq_nam = line[13:-3]
+            elif line[2:11] == 'COLUMN ID' :  # The Contact Pair Info              
+               ContNameHeader.append(line[20:-10])
+            elif line[2:7] == 'UNITS' :       # Reading UNITS Info   
+               pass 
+            elif line[2:-2] == 'HEADER' :     # Last line of the header              
+               nInfoPerPair = len(ContNameHeader)             
+            elif((line[1:8] == 'COLDATA' or line[:3].isspace())):         
+                if(line[1:8] == 'COLDATA'):
+                   #<COLDATA LOAD_STEP="        1" SUBSTEP="        1" ITERATION="        4" TIME="   0.1000000    " PHYSICAL TIME FOR WEAR="   0.1000000    ">   
+                   LoadTime.append(float(line[20:29])) # LOAD_STEP
+                   LoadTime.append(float(line[40:49])) # SUBSTEP 
+                   LoadTime.append(float(line[62:71])) # ITERATION
+                   LoadTime.append(float(line[79:95])) # TIME 
+                   paircount = 0
+                   if(len(line) == 140):
+                        LoadTime.append(float(line[121:136])) # PHYSICAL TIME FOR WEAR
+                        scaletimeflg = 1                        
+                   frq_num = frq_num + 1
+                elif(line[:3].isspace()):                     
+                   nline = list(map(float,line.split()))                                               
+                   ContPairInfo.append(nline) 
+                   paircount = paircount + 1
+            else:
+                pass  
+        #Data in the cnd file is fetched             
+        print( 'Number of contact pair: ' , paircount)
+        print( 'Number information per contact pair: ' , nInfoPerPair)
+        print('Contact data is written  for  ', frq_num , '   ', frq_nam+'S')
+    if(contsrno > paircount):
+       sys.exit('contact serial number ', contsrno, 'should be less than ', paircount)

```

Comment:
Again, best to raise rather than exit. That will give the user the ability to modify their solution rather than just restart python.

```suggestion
    if contsrno > paircount:
       raise RuntimeError(f'Contact serial number {contsrno} should be less than {paircount}')
```